### PR TITLE
Sm controller

### PR DESCRIPTION
**Pull request by shubh90**

### CODE REVIEW

File: proto/controller.proto [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+option go_package = "github.com/google/kne/proto/controller";
+
+// Topology manager service definition.
+service TopologyManager {
+  rpc CreateTopology (CreateTopologyRequest) returns(CreateTopologyResponse) {}
+}
+
+// Request message to create a topology from a file.
+message CreateTopologyRequest {
+  string file_name = 1;
```
Comment (marcushines) [inline proto/controller.proto:27]: how is that going to work... the cli will have the file locally how are you getting the file onto the service running in k8s?

File: proto/controller.proto [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+}
+  
+// Returns create topology response.
+message CreateTopologyResponse {
+  string message = 1;
```
Comment (marcushines) [inline proto/controller.proto:32]: what is a message?

File: proto/controller.proto (hunk 1 above)
Comment (shubh90) [reply proto/controller.proto]: Changed CreateTopologyRequest to take in topology proto

File: proto/controller.proto (hunk 2 above)
Comment (shubh90) [reply proto/controller.proto]: doing empty response right now. But, was also considering CreateTopologyResponse to send topology proto back.

File: proto/controller.proto [hunk 3]
```diff
@@ -22,13 +22,47 @@ option go_package = "github.com/google/kne/proto/controller";
...
+  rpc DeleteTopology (DeleteTopologyRequest) returns(DeleteTopologyResponse) {}
+  rpc ShowTopology (ShowTopologyRequest) returns(ShowTopologyResponse) {}
 }
 
-// Request message to create a topology from a file.
+enum NodeState {
+  STATE_UNKNOWN = 0;
+  STATE_FAILED = 1;
+  STATE_RUNNING = 2;
+  STATE_DELETED = 3;
```
[the remark this answers is not in the document]
Comment (shubh90) [reply proto/controller.proto]: Updated states to: CREATING -> CONTOLLER_AWAIT -> STARTING -> RUNNING -> ERROR

File: proto/controller.proto [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  rpc CreateTopology (CreateTopologyRequest) returns(CreateTopologyResponse) {}
+}
+
+// Request message to create a topology from a file.
+message CreateTopologyRequest {
+  topo.Topology topology = 1;
+}
+  
+// Returns create topology response.
+message CreateTopologyResponse {
```
[the remark this answers is not in the document]
Comment (shubh90) [reply proto/controller.proto:43]: We return the state and the name of the topology created

**Review by marcushines — APPROVED** [review]

**Pull request merged**